app/retail/views.py

```diff
@@ -1089,17 +1089,53 @@ def results(request, keyword=None):
 def activity(request):
     """Render the Activity response."""
     page_size = 15
+    page = int(request.GET.get('page', 1))
+    what = request.GET.get('what', 'everywhere')
+
+    # create diff filters
     activities = Activity.objects.all().order_by('-created_on')
+    if request.user.is_authenticated:
+        relevant_profiles = []
+        if what == 'tribes':
+            from dashboard.models import get_my_earnings
+            relevant_profiles = get_my_earnings(request.user.profile.pk)
+        if 'keyword-' in what:
+            keyword = what.split('-')[1]
+            relevant_profiles = Profile.objects.filter(keywords__icontains=keyword)
+        if 'activity:' in what:
+            pk = what.split(':')[1]
+            activities = activities.filter(pk=pk)
+            if page > 1:
+                activities = Activity.objects.none()
+        # filters
+        if len(relevant_profiles):
+            activities = activities.filter(profile__in=relevant_profiles)
+
+    # after-pk filters
+    if request.GET.get('after-pk'):
+        activities = Activity.objects.filter(pk__gt=request.GET.get('after-pk'))
```

Comment:
Can we use creation date type filter here instead? Using a PK to infer ordering in time can cause problems down the line